serverless/src/main.rs

```diff
@@ -47,6 +49,21 @@ pub async fn main() {
     let level_filter = log::LevelFilter::from_str(&log_level).unwrap_or(log::LevelFilter::Info);
     Builder::new().filter_level(level_filter).init();
 
+    let (_, env_type) = match read_cloud_env() {
+        Some(value) => value,
+        None => {
+            error!("Unable to identify environment. Shutting down Mini Agent.");
+            return;
+        }
+    };
+
+    let dogstatsd_tags = match env_type {
+        EnvironmentType::CloudFunction => "origin:cloudfunction",
+        EnvironmentType::AzureFunction => "origin:azurefunction",
+        EnvironmentType::AzureSpringApp => "origin:azurespringapp",
+        EnvironmentType::LambdaFunction => "origin:lambda", // historical reasons
+    };
```

Comment:
Can we compare this to what we get in other environments (using go-agent)? Do we get `origin`, `dd.origin` or both?

Reply:
We get both `dd.origin` and `origin` tags on DogStatsD metrics for serverless-init/sidecar.

- [Azure Container Apps](https://github.com/DataDog/datadog-agent/blob/db8c605ceb8e620125181ddc758df1bdc8050507/cmd/serverless-init/cloudservice/containerapp.go#L73-L74)
- [Azure Web Apps - Container deployment](https://github.com/DataDog/datadog-agent/blob/db8c605ceb8e620125181ddc758df1bdc8050507/cmd/serverless-init/cloudservice/appservice.go#L38-L39)
- [Google Cloud Run](https://github.com/DataDog/datadog-agent/blob/db8c605ceb8e620125181ddc758df1bdc8050507/cmd/serverless-init/cloudservice/cloudrun.go#L71-L72)

<img width="754" alt="Screenshot 2025-03-28 at 2 53 36 PM" src="https://github.com/user-attachments/assets/7c6e07bd-a78f-40e1-ba76-97085d45a14a" />

Reply:
Let's add both in this case as well for consistency

Reply:
Added `dd.origin` as a tag in addition to `origin`.

https://github.com/DataDog/libdatadog/blob/f9c29f40c44eee6c74fdf0979c6805c4af289cae/serverless/src/main.rs#L61-L64

<img width="957" alt="Screenshot 2025-03-31 at 10 16 53 AM" src="https://github.com/user-attachments/assets/684f8d7e-5571-4bb1-bd66-bd9cf4f7d13e" />